tutorial/CMakeLists.txt

```diff
@@ -48,39 +48,43 @@ add_tutorial(lesson_08_scheduling_2.cpp WITH_IMAGE_IO WITH_OPENMP)
 add_tutorial(lesson_09_update_definitions.cpp WITH_IMAGE_IO WITH_OPENMP)
 
 if (TARGET_PTX)
-    # Tutorial 10 requires that we build generation code, then run it,
-    # so we can build the final executable.
-    add_tutorial(lesson_10_aot_compilation_generate.cpp)
-
-    # LLVM may leak memory during Halide compilation. If projects are built with address sanitizer enabled,
-    # this may cause generators to fail, making it hard to use Halide and address sanitizer at the same time.
-    # To work around this, we execute generators with an environment setting to disable leak checking.
-    set(FILTER_LIB "lesson_10_halide${CMAKE_STATIC_LIBRARY_SUFFIX}")
-    add_custom_command(OUTPUT lesson_10_halide.h "${FILTER_LIB}"
-                       DEPENDS lesson_10_aot_compilation_generate
-                       COMMAND ${CMAKE_COMMAND} -E env "ASAN_OPTIONS=detect_leaks=0" $<TARGET_FILE:lesson_10_aot_compilation_generate>)
-    add_custom_target(exec_lesson_10_aot_compilation_generate
-                      DEPENDS lesson_10_halide.h "${FILTER_LIB}")
-
-    # This will be linked with the code generated by
-    # the generator (lesson_10_aot_compilation_generate)
-    add_executable(lesson_10_aot_compilation_run lesson_10_aot_compilation_run.cpp)
-    add_dependencies(lesson_10_aot_compilation_run exec_lesson_10_aot_compilation_generate)
-    target_link_libraries(lesson_10_aot_compilation_run PRIVATE
-                          "${CMAKE_CURRENT_BINARY_DIR}/${FILTER_LIB}"
-                          Halide::Runtime
-                          Threads::Threads
-                          ${CMAKE_DL_LIBS})
-
-    # Undocumented function in HalideGeneratorHelpers. Do not call in external code.
-    # Users of the AOT functions (as opposed to Generators) should link to the relevant
-    # GPU libraries manually.
-    _Halide_target_link_gpu_libs(lesson_10_aot_compilation_run PRIVATE ${HL_TARGET})
-
-    target_include_directories(lesson_10_aot_compilation_run PRIVATE "${CMAKE_CURRENT_BINARY_DIR}")
-
-    add_test(NAME tutorial_lesson_10_aot_compilation_run COMMAND lesson_10_aot_compilation_run)
-    set_tests_properties(tutorial_lesson_10_aot_compilation_run PROPERTIES LABELS tutorial)
+    if (TARGET_WEBASSEMBLY AND HL_TARGET MATCHES "wasm")
+        # TODO: Requires custom build rules to work under wasm.
```

Comment:
Should we just error out here?

Reply:
No, because you might want to build the _other_ tutorials.

Reply:
An author warning might be appropriate, however, via `message(AUTHOR_WARNING "Not all tutorials build under WASM")` or similar

Reply:
Warnings inserted.